php/src-generated/TestRunStarted.php

```diff
@@ -25,6 +25,7 @@ final class TestRunStarted implements JsonSerializable
      *
      */
     public function __construct(
+        public readonly string $id = '',
```

Comment:
This is a breaking change due to the order of arguments. A call like this would previously have worked:

```php
$t = new TestRunStarted(new TimeStamp());
```
but now will not (because a timestamp is not a string)

For this reason new fields must be added to the end of the list if at all. I remember @mpkorstanje saying the issue is more complex for Java constructors

Reply:
Two things:

- Even appending to the end of the ctor is still a breaking change for Java
- It’s fundamentally a breaking change anyway because this new field is required, so I don’t think the order matters?

Reply:
I think it's mostly a matter of putting the entry in the right place so we know to bump the major version.

Reply:
I could be wrong but adding a required field breaks the ability to decode older messages - without explicit versioning in the streams that could be an issue

Reply:
You're right, in Java at least it will throw if a required field isn't present.

Would it then be better to introduce these fields as nominally optional? We could consider marking them required sometime in the future once they are consistently implemented.

Reply:
☝️ I've done this.